tests/async/test_dialog.py

```diff
@@ -22,64 +21,69 @@
 
 
 async def test_should_fire(page: Page, server):
-    on_dialog_actions_complete: Future[None] = asyncio.Future()
+    result = []
 
-    async def on_dialog(dialog: Dialog):
+    def on_dialog(dialog: Dialog):
+        result.append(True)
         assert dialog.type == "alert"
         assert dialog.defaultValue == ""
         assert dialog.message == "yo"
-        on_dialog_actions_complete.set_result(await dialog.accept())
+        asyncio.create_task(dialog.accept())
 
     page.on("dialog", lambda dialog: asyncio.create_task(on_dialog(dialog)))
     await page.evaluate("alert('yo')")
-    await on_dialog_actions_complete
+    assert result
 
 
 async def test_should_allow_accepting_prompts(page: Page, server):
-    on_dialog_actions_complete: Future[None] = asyncio.Future()
+    result = []
 
-    async def on_dialog(dialog: Dialog):
+    def on_dialog(dialog: Dialog):
+        result.append(True)
         assert dialog.type == "prompt"
         assert dialog.defaultValue == "yes."
         assert dialog.message == "question?"
-        on_dialog_actions_complete.set_result(await dialog.accept("answer!"))
+        asyncio.create_task(dialog.accept("answer!"))
 
     page.on("dialog", lambda dialog: asyncio.create_task(on_dialog(dialog)))
```

Comment:
You are creating two asyncio tasks once the dialog event will be received. Only one would be needed. We try to only wrap as less functions inside a task. So I would do it like shown here.
```suggestion
    page.on("dialog", on_dialog)
```

Reply:
🤦 Thank you! That had meant to go with these cleanups! Fixed in latest commit—at least it should be :)